pkg/api/authz.go

```diff
@@ -133,9 +143,37 @@ func (ac *AccessController) getContext(username string, request *http.Request) c
 	return ctx
 }
 
+func getUserGroup(ac *AccessController, username string) string {
+	for groupName, group := range ac.Config.Groups {
+		for _, user := range group.Users {
+			// find if the user is part of any groupsS
+			if user == username {
+				return groupName
+			}
+		}
+	}
+
+	return ""
+}
+
 // isPermitted returns true if username can do action on a repository policy.
-func isPermitted(username, action string, policyGroup config.PolicyGroup) bool {
+func isPermitted(ac *AccessController, username, action string, policyGroup config.PolicyGroup) bool {
```

Comment:
Any specific reason why this function is not a member of the AccessController struct? Since it needs an AccessController parameter anyway.
Same goes for `getUserGroup`